### ISSUE TITLE
singleton-comparison doesn't cover `expr != [None|True|False]`

### ISSUE THREAD

**Issue by peter50216** [PyCQA/pylint#2037]
### Steps to reproduce
```py
def something():
    a = 1
    if a == None:
        print 'is None'
    if a != None:
        print 'is not None'
```

### Current behavior
```
C:  3, 7: Comparison to None should be 'expr is None' (singleton-comparison)
```

### Expected behavior
```
C:  3, 7: Comparison to None should be 'expr is None' (singleton-comparison)
C:  5, 7: Comparison to None should be 'expr is not None' (singleton-comparison)
```

### pylint --version output
```
No config file found, using default configuration
pylint 1.8.4, 
astroid 1.6.3
Python 2.7.14 (default, Sep 30 2017, 15:47:39) 
[GCC 4.9.4]
```

**Comment by peter50216** [comment]
It seems that the patch doesn't cover the case `None != x`?
